form/form_collections.rst

```diff
@@ -655,15 +655,8 @@ the relationship between the removed ``Tag`` and ``Task`` object.
         use Doctrine\Common\Collections\ArrayCollection;
 
         // ...
-        public function edit($id, Request $request)
+        public function edit(Task $task, Request $request, EntityManagerInterface $entityManager)
         {
-            $entityManager = $this->getDoctrine()->getManager();
```

Comment:
Personally I like removing this code ... but to not complicate the example with more concepts (ParamConverters), let's go back to the original code, but improved with the injection of $em that you proposed. Something like this:

```php
public function edit($id, Request $request, EntityManagerInterface $entityManager)
{
    if (null === $task = $entityManager->getRepository(Task::class)->find($id)) {
        throw $this->createNotFoundException('No task found for id '.$id);
    }
    
    // ...
}
```